File: quickSort.py

def partition (arr,l,u):
    pivot = arr[u]
    idx = l-1
    for i in range(l,u):
        if (arr[i]<pivot):
            (arr[idx+1],arr[i]) = (arr[i],arr[idx+1])
            idx = idx + 1
    (arr[u],arr[idx+1])=(arr[idx+1],arr[u])
    return idx+1

def quickSort(arr,l,u):
    if(l<u):
        pivot = partition(arr,l,u)

        quickSort(arr,l,pivot-1)
        quickSort(arr,pivot+1,u)



arr = [55, 84, 44, 11, 8, 7, 8687, 5, 9, 67, 43, 34]

quickSort(arr,0,len(arr)-1)
print("Sorted array is:")
print(arr)

File: Quick_Sort.cpp
#include <iostream>
using namespace std;
void swap(int* a, int* b) 
{ 
     int t = *a; 
     *a = *b; 
     *b = t; 
} 
int partition (int arr[], int low, int high) 
{ 
    int pivot = arr[high],i=(low-1);
    for (int j = low; j <= high- 1; j++) 
    {
        if (arr[j] <= pivot) 
        { 
            i++;
            swap(&arr[i], &arr[j]); 
        } 
    } 
    swap(&arr[i + 1], &arr[high]); 
    return (i + 1); 
}
void quickSort(int arr[], int low, int high) 
{ 
    if (low < high) 
    { 
        int pivot = partition(arr, low, high); 
        quickSort(arr, low, pivot - 1); 
        quickSort(arr, pivot + 1, high); 
    } 
} 
void displayArray(int arr[], int size) 
{ 
    int i; 
    for (i=0; i < size; i++) 
        cout<<arr[i]<<"\t"; 
} 
int main() 
{ 
    int arr[] = {12,23,3,43,51,35,19,45}; 
    int n = sizeof(arr)/sizeof(arr[0]); 
    cout<<"Input array"<<endl;
    displayArray(arr,n);
    cout<<endl;
    quickSort(arr, 0, n-1); 
    cout<<"Array sorted with quick sort"<<endl; 
    displayArray(arr,n); 
    return 0; 
}
